hw/bsp/family.cmake

```diff
@@ -0,0 +1,61 @@
+if (NOT FAMILY_MCUS)
+    set(FAMILY_MCUS ${FAMILY})
+endif()
+
+# save it in case of re-inclusion
+set(FAMILY_MCUS ${FAMILY_MCUS} CACHE INTERNAL "")
+
+function(family_filter RESULT DIR)
+    get_filename_component(DIR ${DIR} ABSOLUTE BASE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
+    file(GLOB ONLYS "${DIR}/.only.MCU_*")
+    if (ONLYS)
+        foreach(MCU IN LISTS FAMILY_MCUS)
+            if (EXISTS ${DIR}/.only.MCU_${MCU})
+                set(${RESULT} 1 PARENT_SCOPE)
+                return()
+            endif()
+        endforeach()
+    else()
+        foreach(MCU IN LISTS FAMILY_MCUS)
+            if (EXISTS ${DIR}/.skip.MCU_${MCU})
+                set(${RESULT} 0 PARENT_SCOPE)
+                return()
+            endif()
+        endforeach()
+    endif()
+    set(${RESULT} 1 PARENT_SCOPE)
+endfunction()
+
+function(family_add_subdirectory DIR)
+    family_filter(SHOULD_ADD "${DIR}")
+    if (SHOULD_ADD)
+        add_subdirectory(${DIR})
+    endif()
+endfunction()
+
+function(family_get_project_name OUTPUT_NAME DIR)
+    get_filename_component(SHORT_NAME ${DIR} NAME)
+    if (TINYUSB_FAMILY_PROJECT_NAME_INCLUDES_BOARD OR NOT DEFINED TINYUSB_FAMILY_PROJECT_NAME_INCLUDES_BOARD)
+        set(${OUTPUT_NAME} ${TINYUSB_FAMILY_PROJECT_NAME_PREFIX}${BOARD}-${SHORT_NAME} PARENT_SCOPE)
+    else()
+        set(${OUTPUT_NAME} ${TINYUSB_FAMILY_PROJECT_NAME_PREFIX}${SHORT_NAME} PARENT_SCOPE)
```

Comment:
yeah in the SDK build i skip the board name... i assumed you guys wanted it when building from build-family.py, but if not yes we can remove

Reply:
> bsp/family.cmake and bsp/family/rp2040/family.cmake are easily mistaken to each other. We should either rename it to bsp/family_common.cmake or move it to examples/example.cmake. You could give it any other name as well.

Yes i was thinking the same thing... family_common.cmake is probably better, since we are perhaps now using these for things other than examples (i.e. they would be used by the SDK tinyusb cmake build for peoples' own tinyusb based code too)

Reply:
> yeah in the SDK build i skip the board name... i assumed you guys wanted it when building from build-family.py, but if not yes we can remove

Ok, please just remove BOARD from the project name, once this is merged, i will update the gnumake build to match as well.

> Yes i was thinking the same thing... family_common.cmake is probably better, since we are perhaps now using these for things other than examples (i.e. they would be used by the SDK tinyusb cmake build for peoples' own tinyusb based code too)

Yeah, please do

Reply:
i renamed it to `family_common.cmake`

Reply:
thanks, it is resolved